src/resource/archive/ArchiveManager.cpp

```diff
@@ -118,8 +122,15 @@ size_t ArchiveManager::RemoveArchive(std::shared_ptr<Archive> archive) {
     return RemoveArchive(archive->GetPath());
 }
 
-void ArchiveManager::SetArchives(const std::vector<std::shared_ptr<Archive>>& archives) {
-    mArchives = archives;
+void ArchiveManager::SetArchives(std::shared_ptr<std::vector<std::shared_ptr<Archive>>> archives) {
+    mArchives.clear();
+
+    if (archives) {
+        for (const auto& archive : *archives) {
+            mArchives.push_back(archive);
+        }
+    }
```

Comment:
@Kenix3 is this what you were thinking here?

Reply:
This looks exactly like what we were thinking. It looks like we decided to pass lists around into and out of LUS as shared pointers.